src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,70 @@
 package core.basesyntax;
 
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+
 public class WorkWithFile {
-    public void getStatistic(String fromFileName, String toFileName) {
+    private static final int SUPPLY_INDEX = 0;
+    private static final int BUY_INDEX = 1;
+    private static final int RESULT_INDEX = 2;
+    private static final int RECORD_NAME_INDEX = 0;
+    private static final int RECORD_VALUE_INDEX = 1;
+    private static final String SEPARATOR = ",";
+
+    public static void getStatistic(String fromFileName, String toFileName) {
+        List<String> records = readCsvToList(fromFileName);
+        String report = generateReport(records);
+        writeToFile(report, new File(toFileName));
+    }
+
+    private static String generateReport(List<String> records) {
+        int[] totalSupplyBuyResult = calculateSupplyAndBuy(records);
+        totalSupplyBuyResult[RESULT_INDEX] = totalSupplyBuyResult[SUPPLY_INDEX]
+                - totalSupplyBuyResult[BUY_INDEX];
+
+        return new StringBuilder().append("supply,").append(totalSupplyBuyResult[SUPPLY_INDEX])
+                .append(System.lineSeparator())
+                .append("buy,").append(totalSupplyBuyResult[BUY_INDEX])
+                .append(System.lineSeparator())
+                .append("result,").append(totalSupplyBuyResult[RESULT_INDEX])
+                .append(System.lineSeparator()).toString();
+    }
+
+    private static int[] calculateSupplyAndBuy(List<String> records) {
+        int[] resultArray = new int[3];
```

Comment:
All magic numbers, strings, symbols in your code should be constants.
[this article](https://mate-academy.github.io/style-guides/java/java.html#s5.2.4-constant-names)